polyply/src/gen_coords.py

```diff
@@ -31,77 +31,76 @@
 from .annotate_ligands import AnnotateLigands, parse_residue_spec, _find_nodes
 from .build_file_parser import read_build_file
 
-def split_residues(molecule, split):
-    max_resid = len(molecule.nodes)
-    for split_string in split:
-        max_resid = molecule.split_residue(split_string, max_resid)
-
-def gen_coords(args):
-    # Read in the topology
-    topology = Topology.from_gmx_topfile(name=args.name, path=args.toppath)
-    topology.preprocess()
-
+def find_starting_node_from_spec(topology, start_nodes):
+    start_dict = {mol_idx:None for mol_idx, _ in enumerate(topology.molecules)}
+    for start in start_nodes:
+        res_spec = parse_residue_spec(start)
+        if 'mol_idx' in res_spec:
+            mol_idx = res_spec['mol_idx']
+            node = list(_find_nodes(topology.molecules[mol_idx], res_spec))[0]
+            start_dict[mol_idx] = node
+        else:
+            for idx, molecule in enumerate(topology.molecules):
+                if molecule.mol_name == res_spec['molname']:
+                    node = list(_find_nodes(molecule, res_spec))[0]
+                    start_dict[idx] = node
```

Comment:
If mol_idx does not match molname?

Reply:
the else is only entered if there is no mol_idx defined. That is you select any molecule with the name. In case of the if statement mol_idx takes priority over mol-name. Not sure cross-examination is necessary. Does that answer the question?

Reply:
Jup. I'm not sure making sure the input is consistent is worth the effort. I'll leave that up to you

Reply:
Let's leave it for later